### PR TITLE
Changing Start/Stop Request body to optional for Workloads

### PR DESCRIPTION
**Pull request by jaskisin**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

**Review by ajaygit158 — APPROVED** [review]

**Comment by jaskisin** [comment]
This pull request involves updates to the `SAPVirtualInstance` management interfaces and JSON specifications to enhance the documentation and make request bodies optional. The most important changes include adding documentation annotations to request bodies and updating JSON schema requirements.

The changes in this PR address issues identified during testing with the 2024-09-01 API. This API is not yet available to the public.

**Comment by pcsin** [comment]
The change is a bug fix of the original PR - [https://github.com/Azure/azure-rest-api-specs/pull/30292](https://github.com/Azure/azure-rest-api-specs/pull/30292). The PR is reporting SDK generation failures in .NET, but since it was not a blocker for the parent PR as well, approving this bug fix.

---
In reply to: [2327991005](https://github.com/Azure/azure-rest-api-specs/pull/30410#issuecomment-2327991005) [](http://example.com/codeflow?ancestors=2327991005)

**Review by pcsin — APPROVED** [review]
:shipit:

File: specification/workloads/Workloads.SAPVirtualInstance.Management/SAPApplicationServerInstance.tsp [hunk 1]
```diff
@@ -60,7 +60,11 @@ interface SAPApplicationServerInstances {
    */
   start is ArmResourceActionAsync<
     SAPApplicationServerInstance,
-    StartRequest,
+    {
+      @doc("SAP Application server instance start request body.")
+      @body
+      body?: StartRequest;
```
Comment (MaryGao) [inline specification/workloads/Workloads.SAPVirtualInstance.Management/SAPApplicationServerInstance.tsp:66]: Could you help me understand why this POST body could be optional?

Comment (weidongxu-microsoft) [reply specification/workloads/Workloads.SAPVirtualInstance.Management/SAPApplicationServerInstance.tsp:66]: As far as I know, no guideline about this.

There could be valid POST that perform an action, even without a request body; but given request body, the action could be performed a bit differently.

Comment (jaskisin) [reply specification/workloads/Workloads.SAPVirtualInstance.Management/SAPApplicationServerInstance.tsp:66]: Hello @MaryGao @weidongxu-microsoft , yes, in service, we have some default execution of StartRequest when body is not passed. And execution changes when options for StartRequest is passed in the body.

Comment (MaryGao) [reply specification/workloads/Workloads.SAPVirtualInstance.Management/SAPApplicationServerInstance.tsp:66]: Got it and thanks all!

**Pull request merged**